common-custom-user-data-maven-extension/src/main/java/com/gradle/maven/common/user/data/extension/ApiAccessor.java

```diff
@@ -0,0 +1,72 @@
+package com.gradle.maven.common.user.data.extension;
+
+import org.apache.maven.MavenExecutionException;
+import org.codehaus.plexus.PlexusContainer;
+import org.codehaus.plexus.classworlds.realm.ClassRealm;
+import org.codehaus.plexus.logging.Logger;
+
+import java.util.Optional;
+
+import static java.util.Comparator.comparing;
+
+final class ApiAccessor {
```

Comment:
Are we able to provide this via the Plexus container from the Maven extension itself such that this code is handled on our side and then injected into the custom event spy?

Reply:
no, because this code is especially about sharing things from the Maven extension to other places.

The only way to do that would be to hack parent realms from the maven extension side.
Basically, store in 'plexus.core' or another realm that is visible to all extensions in all cases the things we want to be visible to all extensions.
That would be possible I guess (it's another version of Marc's hack) but really not what Maven design wants us to do. (Marc's hack is about [this bug](https://issues.apache.org/jira/browse/MNG-6906) but does not violate Maven's principles, because if we understood exported packages properly, it should have worked even without the hack in all cases)

Then, even if we were ok doing this, we would still have to do the other way around: make the `GradleEnterpriseListener` implementation on the consuming extension available on the Maven extension side:
(this guy:
```
importPackage(extensionRealm, maybeGERealm.get(), OWN_PACKAGE);
```
)

Reply:
Spent many hours on this, and finally got something. Which made the consuming part (i.e. the CCUDME) much nicer and shorter.